### PR TITLE
add SVG rules

### PR DESCRIPTION
**Pull request by 3den**
https://github.com/yahoo/atomizer/issues/209#issuecomment-121396276 @src-code @renatoi @thierryk

### CODE REVIEW

File: src/rules.js [hunk 1]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
         }]
+    },
+    /**
+    ==================================================================
+    SVG
+    ==================================================================
+    */
+    {
+        "type": "pattern",
+        "name": "fill",
```
Comment (src-code) [inline src/rules.js]: The 'name' field is used when we generate the reference docs, so it needs to be formatted.  I'd go with something like "Fill (SVG)", etc for these

File: src/rules.js [hunk 2]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
+        },
+        "arguments": [colors]
+    },
+    {
+        "type": "pattern",
+        "name": "stroke",
+        "matcher": "Stkw",
+        "allowParamToValue": true,
+        "styles": {
+            "stroke": "$0"
```
Comment (src-code) [inline src/rules.js]: Should be `stroke-width`

File: src/rules.js [hunk 3]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
+        "matcher": "Stk",
+        "allowParamToValue": true,
+        "styles": {
+            "stroke": "$0"
+        },
+        "arguments": [colors]
+    },
+    {
+        "type": "pattern",
+        "name": "stroke",
```
Comment (src-code) [inline src/rules.js]: Should be "Stroke width (SVG)"

File: src/rules.js [hunk 4]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
+            "b": "butt",
+            "r": "round",
+            "s": "square"
+        }]
+    },
+    {
+        "type": "pattern",
+        "name": "stroke-linejoin",
+        "matcher": "Stklj",
+        "allowParamToValue": true,
```
Comment (src-code) [inline src/rules.js]: Is this accurate?  I don't think it allows arbitrary values... https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linejoin

File: src/rules.js [hunk 5]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
+        "styles": {
+            "fill": "$0"
+        },
+        "arguments": [colors]
+    },
+    {
+        "type": "pattern",
+        "name": "stroke",
+        "matcher": "Stk",
+        "allowParamToValue": true,
```
Comment (src-code) [inline src/rules.js]: Is this one limited only to the values in `colors`?  If so, I think `allowParamToValue` should be false

File: src/rules.js [hunk 6]
```diff
@@ -2687,5 +2687,72 @@ module.exports = [
...
+        },
+        "arguments": [{
+            "i": "inherit"
+        }]
+    },
+    {
+        "type": "pattern",
+        "name": "stroke-linecap",
+        "matcher": "Stklc",
+        "allowParamToValue": true,
```
Comment (src-code) [inline src/rules.js]: I think this should be `false`: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke-linecap

File: src/rules.js (hunk 5 above)
Comment (3den) [inline src/rules.js]: i think stroke have have more than just the color, seems like you can specify all stroke props with this rule

Comment (src-code) [inline src/rules.js]: https://developer.mozilla.org/en-US/docs/Web/SVG/Attribute/stroke

It seems to only accept a "paint" value, same as `fill`?

**Comment by src-code** [comment]
So following up, `stroke` and `fill` take "paint" values, which could be a color or one of a few keywords or references to other defined gradients, etc:

http://www.w3.org/TR/SVG/painting.html#SpecifyingPaint

If we allow param-to-value as we currently do, then I guess we'd be covered on most of these (the `funciri` might be an issue though)  Or we can list out these keywords as valid arguments, and then disable `allowParamToValue` to restrict people from passing invalid values.

**Comment by 3den** [comment]
@src-code can we pass custom variables is allowParamToValue is set to false?

**Comment by src-code** [comment]
@3den Yes - `allowParamToValue` simply controls whether you can pass an a value such as `1em`, `100px`, etc.

**Comment by 3den** [comment]
Got it, i will fix that

**Comment by 3den** [comment]
@src-code fixed

**Comment by src-code** [comment]
@3den So if we disable `allowParamToValue` on those, we won't be able to pass valid Paint keywords like `none` and `currentColor`, I believe, so we'd have to add those explicitly to the `arguments` array on those rules.

**Comment by 3den** [comment]
@src-code i think you already have those in colors (https://github.com/yahoo/atomizer/blob/master/src/colors.js)

**Comment by src-code** [comment]
@3den Oooh, I didn't see `currentColor` in there!  But's missing `none`, so we'd have to add that one.

**Comment by src-code** [comment]
@3den :+1:

**Pull request merged**